TypeCobol.Analysis/Cfg/ControlFlowGraphBuilder.PerformTarget.cs

```diff
@@ -0,0 +1,95 @@
+            /// <summary>
+            /// All target procedures.
+            /// </summary>
+            internal List<Procedure> Procedures { get; private set; }
+
+            public PerformTarget(List<Sentence> sentences, List<Procedure> procedures)
+            {
+                this.Sentences = sentences;
+                this.Procedures = procedures;
+            }
+        }
+
+        /// <summary>
+        /// Cache of PerformTarget instances already calculated for a Perform procedure 
+        /// </summary>
+        private Dictionary<PerformProcedure, PerformTarget> _performTargetCache;
+        private PerformTarget ComputePerformTarget(PerformProcedure p, SectionNode sectionNode)
+        {
+            if (_performTargetCache == null)
+                _performTargetCache = new Dictionary<PerformProcedure, PerformTarget>();
+            if (_performTargetCache.TryGetValue(p, out var target)) {
+                return target;
+            }
+
+            SymbolReference procedureReference = p.CodeElement.Procedure;
+            SymbolReference throughProcedureReference = p.CodeElement.ThroughProcedure;
+
+            Node procedureNode = ResolveProcedure(p, sectionNode, procedureReference);
+            if (procedureNode == null)
+                return null;
+
+            Procedure procedure = _nodeToProcedure[procedureNode];
+            List<Sentence> sentences = new List<Sentence>();
+            List<Procedure> procedures = new List<Procedure>();
+            if (throughProcedureReference != null)
+            {
+                Node throughProcedureNode = ResolveProcedure(p, sectionNode, throughProcedureReference);
+                if (throughProcedureNode == null)
+                    return null;
+                
+                Procedure throughProcedure = _nodeToProcedure[throughProcedureNode];
+                if (procedure.Number > throughProcedure.Number)
+                {
+                    // the second procedure name is declared before the first one.
+                    this.Cfg.AddWrongOrderPerformThru(p, procedureNode, throughProcedureNode);
+                    return null;
+                }
+
+                //Accumulate sentences located between the two procedures
+                int n = 0;
+                int currentProcedureNumber = procedure.Number;
+                while (currentProcedureNumber <= throughProcedure.Number)
+                {
+                    var currentProcedure = this.CurrentProgramCfgBuilder.AllProcedures[currentProcedureNumber];
+                    currentProcedure.AccumulateSentencesThrough(sentences, throughProcedure, out var lastProcedure);
+                    currentProcedureNumber = lastProcedure.Number + 1;
+                    procedures.Add(currentProcedure);
+                    while (n < sentences.Count)
+                    {
+                        sentences[n++].Procedure = currentProcedure;
+                    }
+                }
```

Comment:
No need to track the current procedure using the `n` index. The `Procedure` property of each `Sentence` object can (and should) be set at construction time. See my previous remark in this PR to see how to do it.

Reply:
done in 4b969b9